include/SFML/Graphics/Drawable.hpp

```diff
@@ -49,6 +49,36 @@ class SFML_GRAPHICS_API Drawable
     ////////////////////////////////////////////////////////////
     virtual ~Drawable() = default;
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    ////////////////////////////////////////////////////////////
+    Drawable() = default;
```

Comment:
I'm not sure why these changes are necessary. This is not a class you can construct so it's not copy nor move constructible. It is however copy and move assignable. I'm not sure what any of this code or the changes you made in Drawable.test.cpp are accomplishing. I think you can remove it all.

Reply:
Because of the creation of the move constructor, the compiler saw the other constructors as implicitly deleted and thus causing build errors, if I recall correctly.

Reply:
It doesn't matter that you added `= default` to those constructors. This class contains a pure virtual function so it can never been constructible. You can tell this is true because the type traits for `sf::Drawable` did not change after you added all these defaulted functions. The class's functionality has not changed.

I checked out this PR and reverted all the changes to `sf::Drawable` and it built just fine for me.

Reply:
I just tried again to be sure and this is what I get:
```
FAILED: src/SFML/Graphics/CMakeFiles/sfml-graphics.dir/VertexArray.cpp.obj 
C:\PROGRA~1\MICROS~2\2022\COMMUN~1\VC\Tools\MSVC\1436~1.325\bin\Hostx86\x86\cl.exe  /nologo /TP -DSFML_GRAPHICS_EXPORTS -DSTBI_FAILURE_USERMSG -D_CRT_SECURE_NO_WARNINGS -D_WINSOCK_DEPRECATED_NO_WARNINGS -IC:\Users\seppe\CLionProjects\SFML_fork\include -IC:\Users\seppe\CLionProjects\SFML_fork\src -external:IC:\Users\seppe\CLionProjects\SFML_fork\extlibs\headers\stb_image -external:IC:\Users\seppe\CLionProjects\SFML_fork\extlibs\headers\glad\include -external:IC:\Users\seppe\CLionProjects\SFML_fork\extlibs\headers\freetype2 -external:W0 /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /WX /W4 /w14242 /w14254 /w14263 /w14265 /w14287 /we4289 /w14296 /w14311 /w14545 /w14546 /w14547 /w14549 /w14555 /w14619 /w14640 /w14826 /w14905 /w14906 /w14928 /permissive- /wd4068 /wd4505 /wd4800 -std:c++17 /showIncludes /Fosrc\SFML\Graphics\CMakeFiles\sfml-graphics.dir\VertexArray.cpp.obj /Fdsrc\SFML\Graphics\CMakeFiles\sfml-graphics.dir\ /FS -c C:\Users\seppe\CLionProjects\SFML_fork\src\SFML\Graphics\VertexArray.cpp
C:\Users\seppe\CLionProjects\SFML_fork\src\SFML\Graphics\VertexArray.cpp(35): error C2512: 'sf::Drawable': no appropriate default constructor available
C:\Users\seppe\CLionProjects\SFML_fork\include\SFML/Graphics/Drawable.hpp(43): note: see declaration of 'sf::Drawable'
C:\Users\seppe\CLionProjects\SFML_fork\src\SFML\Graphics\VertexArray.cpp(35): note: see reference to function 'sf::VertexArray::VertexArray(void)'
C:\Users\seppe\CLionProjects\SFML_fork\src\SFML\Graphics\VertexArray.cpp(40): error C2512: 'sf::Drawable': no appropriate default constructor available
C:\Users\seppe\CLionProjects\SFML_fork\include\SFML/Graphics/Drawable.hpp(43): note: see declaration of 'sf::Drawable'
[64/105] Building CXX object src\SFML\Graphics\CMakeFiles\sfml-graphics.dir\ConvexShape.cpp.obj
[65/105] Building CXX object src\SFML\Graphics\CMakeFiles\sfml-graphics.dir\RenderTarget.cpp.obj
[66/105] Building CXX object src\SFML\Graphics\CMakeFiles\sfml-graphics.dir\RenderWindow.cpp.obj
[67/105] Building CXX object src\SFML\Graphics\CMakeFiles\sfml-graphics.dir\Sprite.cpp.obj
FAILED: src/SFML/Graphics/CMakeFiles/sfml-graphics.dir/Sprite.cpp.obj 
C:\PROGRA~1\MICROS~2\2022\COMMUN~1\VC\Tools\MSVC\1436~1.325\bin\Hostx86\x86\cl.exe  /nologo /TP -DSFML_GRAPHICS_EXPORTS -DSTBI_FAILURE_USERMSG -D_CRT_SECURE_NO_WARNINGS -D_WINSOCK_DEPRECATED_NO_WARNINGS -IC:\Users\seppe\CLionProjects\SFML_fork\include -IC:\Users\seppe\CLionProjects\SFML_fork\src -external:IC:\Users\seppe\CLionProjects\SFML_fork\extlibs\headers\stb_image -external:IC:\Users\seppe\CLionProjects\SFML_fork\extlibs\headers\glad\include -external:IC:\Users\seppe\CLionProjects\SFML_fork\extlibs\headers\freetype2 -external:W0 /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /WX /W4 /w14242 /w14254 /w14263 /w14265 /w14287 /we4289 /w14296 /w14311 /w14545 /w14546 /w14547 /w14549 /w14555 /w14619 /w14640 /w14826 /w14905 /w14906 /w14928 /permissive- /wd4068 /wd4505 /wd4800 -std:c++17 /showIncludes /Fosrc\SFML\Graphics\CMakeFiles\sfml-graphics.dir\Sprite.cpp.obj /Fdsrc\SFML\Graphics\CMakeFiles\sfml-graphics.dir\ /FS -c C:\Users\seppe\CLionProjects\SFML_fork\src\SFML\Graphics\Sprite.cpp
C:\Users\seppe\CLionProjects\SFML_fork\src\SFML\Graphics\Sprite.cpp(40): error C2512: 'sf::Drawable': no appropriate default constructor available
C:\Users\seppe\CLionProjects\SFML_fork\include\SFML/Graphics/Drawable.hpp(43): note: see declaration of 'sf::Drawable'
C:\Users\seppe\CLionProjects\SFML_fork\src\SFML\Graphics\Sprite.cpp(47): error C2512: 'sf::Drawable': no appropriate default constructor available
C:\Users\seppe\CLionProjects\SFML_fork\include\SFML/Graphics/Drawable.hpp(43): note: see declaration of 'sf::Drawable'
[68/105] Building CXX object src\SFML\Graphics\CMakeFiles\sfml-graphics.dir\Shader.cpp.obj
ninja: build stopped: subcommand failed.
```

Reply:
https://github.com/ChrisThrasher/SFML/actions/runs/5193825019/jobs/9364846312

I removed the changes to `sf::Drawable` and the CI pipeline is happy with it. I think something is weird about your local build environment but I'm not sure what.